components/infra-proxy-service/server/nodes.go

```diff
@@ -191,6 +191,56 @@ func (s *Server) UpdateNode(ctx context.Context, req *request.UpdateNode) (*resp
 	}, nil
 }
 
+// UpdateNodeTags updates the tags
+func (s *Server) UpdateNodeTags(ctx context.Context, req *request.UpdateNodeTags) (*response.UpdateNodeTags, error) {
+	err := validation.New(validation.Options{
+		Target:  "node",
+		Request: *req,
+		Rules: validation.Rules{
+			"OrgId":    []string{"required"},
+			"ServerId": []string{"required"},
+			"Name":     []string{"required"},
+			"Action":   []string{"required"},
+		},
+	}).Validate()
+
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := s.createClient(ctx, req.OrgId, req.ServerId)
+	if err != nil {
+		return nil, err
+	}
+
+	chefNode, err := c.client.Nodes.Get(req.Name)
+	if err != nil {
+		return nil, ParseAPIError(err)
+	}
+
+	tags := SafeSliceFromMap(chefNode.NormalAttributes, "tags")
+	switch req.Action {
+	case "add":
+		tags = Unique(append(tags, req.Tags...))
+	case "delete":
+		tags = SubstractSlice(tags, req.Tags)
+	case "set":
```

Comment:
does add will do append the remaining list and set will create the new set of tag ?

Reply:
Yes, add will be used for appending to existing tags, set will reset/create a new set of tags.